docs/app/references/configuration.mdx

```diff
@@ -589,6 +590,55 @@ The `**/node_modules/**` pattern is automatically added to `excludeSpecPattern`
 and does not need to be specified (and can't be overridden). See [`e2e`](#e2e)
 or [`component`](#component) testing-specific options.
 
+### injectDocumentDomain
+
+This option is <Badge type="caution">deprecated</Badge>, and will be removed in the next
+major version of Cypress.
+
+Set this configuration option to `true` to instruct Cypress to
+[inject document.domain](/app/guides/cross-origin-testing#What-Cypress-does-under-the-hood)
+into your test application. This can reduce the need for `cy.origin()` when [navigating
+between subdomains](/app/guides/cross-origin-testing), but comes with compatibility
+caveats for some sites.
+
+This configuration option is provided to ease the transition between `cy.origin()`'s behavior
+in Cypress 13 and the default behavior in Cypress 14. It will be removed in Cypress 15.
+[Read the Cypress 14 migration guide](/app/references/migration-guide#Migrating-to-Cypress-140) to understand how to update your tests to remove
+the need to set this flag.
+
+This configuration value _must_ be set to true if you are running tests with experimental
+WebKit support, as `cy.origin` is not yet supported in WebKit.
+
+:::caution
+
+#### Known Incompatibilities
+
+Setting this configuration value to `true` can cause the application you are testing
+to behave in unexpected ways. This is especially true if a site your tests visit
```

Comment:
there are implications if your site sets it, but there are also implications where the browser defaults the `origin-agent-cluster` into buckets, which can cause problems. I think this is also worth mentioning